tests/run/cpp_stl_string_view.pyx

```diff
@@ -35,6 +35,25 @@ def test_return_to_py():
     return return_to_py()
 
 
+def test_coercion_py(py_str):
+    """
+    >>> b_asdf_5 = b_asdf * 5
+    >>> test_coercion_py(b_asdf_5)
+    """
+    if not isinstance(py_str, bytes):
+        return
+
+    cdef string_view sv1 = <bytes>py_str
+    assert len(py_str) == sv1.size()
+    assert <const char*>py_str == sv1.data()
+
+    cdef const char[:] mv = py_str
+    cdef string_view sv2 = mv
```

Comment:
Looks like the last assignment breaks down on the C++ side

Wondering if we need to extend some existing Cython logic

This in particular stick out, but maybe there are other code paths worth looking at

https://github.com/cython/cython/blob/bf59121ba5c9f79b4bbfeab05407c15e07e4cb15/Cython/Utility/CppConvert.pyx#L4-L35